tcms/bugs/tests/test_models.py

```diff
@@ -0,0 +1,69 @@
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.test import TestCase
+from django.utils.translation import gettext_lazy as _
+from mock import patch
+
+from tcms.tests.factories import BugFactory, UserFactory
+
+
+class TestSendMailOnAssigneeChange(TestCase):
+    """Test that assignee is notified by mail each time they are assigned a bug.
+
+    Ideally, notifications are sent out when:
+    * Assignee is assigned bug on bug creation
+    * Assignee is assigned bug which was previously assigned to someone other assignee
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.site = Site.objects.get(pk=settings.SITE_ID)
+
+    def _assert_send_mail_called_appropriately(self, send_mail, i, bug, assignee):
+        """
+        Helper to check that send_mail is called with the expected arguments.
+
+        :param send_mail: The patch of the `send_mail` method
+        :param i: The zero based index/call_count of the call being checked
+        :param bug: The bug object being assigned
+        :param assignee: The assignee who is to be notified
+        """
+        # Subject of mail
+        self.assertIn(str(_('New Bug Assignment')), send_mail.call_args_list[i][0][0])
+
+        # Intro in body
+        self.assertIn(
+            str(_('You\'ve been assigned a new bug by {}'.format(bug.reporter))),
+            send_mail.call_args_list[i][0][1]
+        )
+        # Bug summary in body
+        self.assertIn(
+            str(_('{}'.format(bug.summary))),
+            send_mail.call_args_list[i][0][1]
+        )
+        # Bug absolute url in body
+        self.assertIn(
+            'http://{}{}'.format(self.site.domain, bug.get_absolute_url()),
+            send_mail.call_args_list[i][0][1]
+        )
+
+        # Recepient email address
+        self.assertIn(assignee.email, send_mail.call_args_list[i][0][3])
+
+    @patch('tcms.core.utils.mailto.send_mail')
+    def test_notify_assignee_on_bug_creation(self, send_mail):
+        assignee = UserFactory()
+        bug = BugFactory(assignee=assignee)
+
+        self.assertTrue(send_mail.called)
+        self.assertEqual(send_mail.call_count, 1)
+        self._assert_send_mail_called_appropriately(
+            send_mail, send_mail.call_count - 1, bug, assignee
+        )
```

Comment:
Once you make changes as to how context is passed to the email template then you can easily reconstruct the expected subject/body in the test suite and replace this helper assert method with `send_mail.assert_called_once_with()` as seen in TestSendMailOnCaseIsDeleted